index.js

```diff
@@ -53,8 +53,9 @@ var initConfig = function(projectConfig) {
   var config = {};
   util._extend(config, require('./config.js'));
   util._extend(config, projectConfig);
-  if (process.env.hasOwnProperty('GCLOUD_TRACE_CONFIG')) {
-    util._extend(config, require(path.resolve(process.env.GCLOUD_TRACE_CONFIG)));
+  if (process.env.hasOwnProperty('GCLOUD_CONFIG')) {
+    util._extend(config,
+      require(path.resolve(process.env.GCLOUD_CONFIG)).trace);
```

Comment:
This breaks (i.e. throws) when `require` returns a value that is not deference-able, e.g. `null` or `undefined`. You should check that `require` result is truthy.